roofit/roofitcore/src/TestStatistics/RooBinnedL.cxx

```diff
@@ -0,0 +1,163 @@
+// Author: Patrick Bos, Netherlands eScience Center / NIKHEF 2021
+
+/*****************************************************************************
+ * RooFit
+ * Authors:                                                                  *
+ *   WV, Wouter Verkerke, UC Santa Barbara, verkerke@slac.stanford.edu       *
+ *   DK, David Kirkby,    UC Irvine,         dkirkby@uci.edu                 *
+ *                                                                           *
+ * Copyright (c) 2000-2021, Regents of the University of California          *
+ *                          and Stanford University. All rights reserved.    *
+ *                                                                           *
+ * Redistribution and use in source and binary forms,                        *
+ * with or without modification, are permitted according to the terms        *
+ * listed in LICENSE (http://roofit.sourceforge.net/license.txt)             *
+ *****************************************************************************/
+
+/**
+\file RooBinnedL.cxx
+\class RooBinnedL
+\ingroup Roofitcore
+
+Class RooBinnedL implements a a -log(likelihood) calculation from a dataset
+and a PDF. The NLL is calculated as
+<pre>
+ Sum[data] -log( pdf(x_data) )
+</pre>
+In extended mode, a (Nexpect - Nobserved*log(NExpected) term is added
```

Comment:
I tried to improve the documentation *by a lot* in my 2y, so please take inspiration from
https://github.com/root-project/root/blob/55a7841e24ac8a65d92e996f85449c0027424b64/roofit/roofitcore/src/RooNLLVar.cxx#L17-L29

Currently, we would revert to a state before I started improving the docs.